models/llama-2-7b-vllm/predict.py

```diff
@@ -0,0 +1,148 @@
+import asyncio
+import functools
+import io
+import time
+import zipfile
+from typing import Optional, Tuple
+
+import torch
+from cog import BasePredictor, ConcatenateIterator, Input, Path
+
+from config import (LOCAL_DEFAULT_INFERENCE_WEIGHTS_PATH,
+                    REMOTE_DEFAULT_INFERENCE_FILES_TO_DOWNLOAD,
+                    REMOTE_DEFAULT_INFERENCE_WEIGHTS_PATH)
+from src.download import Downloader
+from src.inference_engines.vllm_engine import vLLMEngine
+from src.utils import maybe_download_with_pget
+
+
```

Comment:
So this is broadly great! Ran it locally and it works (with the dependency & import changes mentioned elsewhere), speed is as fast as advertised. It's generally excellent.

I think the one thing left is more of a packaging issue - I'm curious if we can rewrite this code s.t. we're using the existing `predict.py`, perhaps with some modifications to `predict.py` to work with async code. That's the goal with the `Engine` abstraction - it should let us reuse the same `predict.py`. Juggling multiple copies of predict per engine/model/etc. was kind of a nightmare to maintain, so I'd like to avoid returning to that state if possible. 

Happy to pair on that tomorrow (or take a shot at it myself if you prefer), let me know! Also IMO this shouldn't block us testing this code in stage, we can work in parallel here.

Reply:
Resolving this comment since I want to delete this `predict.py`!